braindecode/classifier.py

```diff
@@ -1,16 +1,83 @@
-from skorch.callbacks import EpochTimer, BatchScoring, PrintLog
-from skorch.utils import train_loss_score, valid_loss_score, noop
+import numpy as np
+import torch
+from sklearn.metrics import get_scorer
+from skorch.callbacks import EpochTimer, BatchScoring, PrintLog, EpochScoring
 from skorch.classifier import NeuralNet
 from skorch.classifier import NeuralNetClassifier
-import torch
-from torch.utils.data.dataloader import DataLoader
-import numpy as np
+from skorch.utils import train_loss_score, valid_loss_score, noop
+
+from .scoring import PostEpochTrainScoring, CroppedTrialEpochScoring
 
 
 class EEGClassifier(NeuralNetClassifier):
     """Classifier that does not assume softmax activation.
     Calls loss function directly without applying log or anything.
+
+    Parameters
+    ----------
+    cropped: bool (default=False)
+        Defines whether torch model passed to this class is cropped or not.
+        Currently used for callbacks definition.
+    callbacks: None or list of strings or list of Callback instances (default=None)
+        More callbacks, in addition to those returned by
+        ``get_default_callbacks``. Each callback should inherit from
+        :class:`skorch.callbacks.Callback`. If not ``None``, callbacks can be a
+        list of strings specifying `sklearn` scoring functions (for scoring
+        functions names see: https://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter)
+        or a list of callbacks where the callback names are inferred from the
+        class name. Name conflicts are resolved by appending a count suffix
+        starting with 1, e.g. ``EpochScoring_1``. Alternatively,
+        a tuple ``(name, callback)`` can be passed, where ``name``
+        should be unique. Callbacks may or may not be instantiated.
+        The callback name can be used to set parameters on specific
+        callbacks (e.g., for the callback with name ``'print_log'``, use
+        ``net.set_params(callbacks__print_log__keys_ignored=['epoch',
+        'train_loss'])``).
     """
+    # TODO: Update docstring to use NeuralNetClassifier docstring with some
+    #  imporvements
+
+    def __init__(self, *args, cropped=False, callbacks=None, **kwargs):
+        self.cropped = cropped
+        callbacks = self._parse_callbacks(callbacks)
+
+        super().__init__(*args, callbacks=callbacks, **kwargs)
+
+    def _parse_callbacks(self, callbacks):
+        callbacks_list = []
+        if callbacks is not None:
+            for callback in callbacks:
+                if isinstance(callback, tuple):
+                    callbacks_list.append(callback)
+                else:
+                    assert isinstance(callback, str)
```

Comment:
Can you change to 
```suggestion
                    assert isinstance(callback, str) or callable(callback)
```
This is also what get_scorer expects from scikit learn https://scikit-learn.org/stable/modules/generated/sklearn.metrics.get_scorer.html

Reply:
Ah ok I see you might then have to also change line below with _score_func.name

Reply:
Hm Ok I see you use assumption you have a string quite often below... Then ignore my comment, maybe for a future time I leave it here